Feat/live updates command

vaishvi2811 opened the pull request:
**Description**
This PR introduces the /marketupdate slash command to the Market Bot app. It allows users to instantly fetch real-time prices for both stocks and cryptocurrencies directly within Rocket.Chat.

**Flow**
- User runs: /marketupdate <symbol>
- The bot determines if the symbol is a stock or a crypto asset.
- For cryptocurrencies, it fetches live prices using the CoinGecko API.
 For stocks, it fetches live prices using the Alpha Vantage API (API key managed securely via app settings).
- The bot responds in the chat with the current price or an error message if the symbol is not found.

**Features**
- Unified command: One command for both stocks and crypto tickers.
- Real-time data: Integrates with CoinGecko and Alpha Vantage for up-to-date prices.
- User-friendly feedback: Clear messages for missing symbols or unavailable data.


https://github.com/user-attachments/assets/bafb8c4d-2303-4a9e-be4e-54b5c1d048e1

vaishvi2811 committed:
feat:/marketupdate command for real-time stock and crypto prices

Spiral-Memory commented:
Hi @vaishvi2811 
Can you please resolve conflicts and rebase / update this branch with the main branch

vaishvi2811 committed:
feat: enhance market update & logo-name updated

vaishvi2811 commented:
Hi @Spiral-Memory ! 
I have rebased this branch and have also made some enhancements for richer user-experience as below: 

* Enhanced Data: Provided more detailed information to users about stocks (via Alpha Vantage) and cryptocurrencies like BTC (via CoinGecko), including technical indicators and rich market data.
* Efficient API Usage: Used Alpha Vantage’s GLOBAL_QUOTE endpoint for stocks to maximize efficiency and minimize API calls, and fetched comprehensive crypto data from CoinGecko.
* Branding Update: Updated the app’s name to "FinKit" and added a custom logo for improved branding and user experience.

Attaching a video of the same, looking forward to your review!

https://github.com/user-attachments/assets/acdde61b-c149-4f43-98ae-d33c89f9a1bf

Spiral-Memory commented on market-bot/app.json:
```diff
@@ -2,14 +2,14 @@
     "version": "0.0.1",
     "requiredApiVersion": "1.50.0",
-    "iconFile": "icon.png",
+    "iconFile": "FinKitLogo.png",
```
I think we do need to change the logo; it doesn't really look good.

Spiral-Memory commented on market-bot/src/commands/MarketUpdateCommand.ts:
```diff
@@ -0,0 +1,66 @@
+        let message: string;
+        if (marketData && marketData.price) {
+            if (marketData.isCrypto) {
+                message = `🪙 *${marketData.name}* (${marketData.symbol})\n`
```
Can we define it separately.

Spiral-Memory commented on market-bot/src/external/AlphaVantage.ts:
```diff
@@ -1,60 +1,90 @@
-            rsi: getLastValue(rsiResp.data, 'Technical Analysis: RSI'),
-        };
+        // SMA 50
+        const sma50Url = `https://www.alphavantage.co/query?function=SMA&symbol=${encodeURIComponent(symbol)}&interval=daily&time_period=50&series_type=close&apikey=${apiKey}`;
```
Use env variables or constant.tsx

Spiral-Memory reviewed (approved):
Great work, @vaishvi2811! Thanks for the effort. 🚀 Let's try to make this bot more conversational in your next PR. I've added some minor comments; just fix those, and we are good to merge this.

vaishvi2811 committed:
refactor:constants.ts & message formatting

Amarta113 commented:
It is empty.
and yes it is in `src/external` folder

vaishvi2811 commented:
@Amarta113 The line was present locally but got missed in the merged commit.

For now, you can just add the line I shared above to your constants.ts for now, that should let your app zip and deploy. Then, head to the app settings on the server and add your API key under alpha_vantage_api_key.

Spiral-Memory commented:
Hey @Amarta113 @vaishvi2811
Whoever is available and can do, please raise a minor PR fixing this

Thanks

Amarta113 commented:
@vaishvi2811  I did, but still facing the same issue.

vaishvi2811 commented:
@Amarta113 Please pull the latest changes on your end and try setting up again. I have tested the changes and pushed a fix. Let me know if you still face any error :)

![image](https://github.com/user-attachments/assets/dfa5bc5b-c133-4a0e-9d76-f3c59fb2ee29)

Amarta113 commented:
I updated the repo but now getting new error: 
![image](https://github.com/user-attachments/assets/b51ddf8f-5370-4fdd-8a03-682fc9b60208)

Spiral-Memory commented:
@vaishvi2811
Can we update this app in your free time, or maybe @Amarta113 downgrade your rocket chat server for a while to test this
